app/factory/Layer.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * [extend description]
+ * @type {String}
+ */
+Ext.define('CpsiMapview.factory.Layer', {
+    alternateClassName: 'LayerFactory',
+    requires: [],
+
+    singleton: true,
+
+    /**
+     * Creates an OpenLayer layer object from a JSON config
+     *
+     * @param  {Object} layerConf  The configuration object
+     * @return {ol.layer.Base} OL layer object
+     */
+    createLayer: function(layerConf) {
+
+        var layerType = layerConf.layerType;
+        var mapLayer;
+
+        switch (layerType) {
+        case 'wms':
+            mapLayer = LayerFactory.createWms(layerConf);
+            break;
+        case 'wfs':
+            mapLayer = LayerFactory.createWFS(layerConf);
```

Comment:
We should create an issue wrt to handling abbreviations and casing consistently. Non-blocking of course.

Reply:
Agree, this was transferred by some legacy code.